lib/token.cpp

```diff
@@ -2741,5 +2741,6 @@ bool Token::isCpp() const
     if (mTokensFrontBack && mTokensFrontBack->list) {
         return mTokensFrontBack->list->isCPP();
     }
+    assert(false);
```

Comment:
I have the feeling that a `throw InternalError` would be nicer than an assert. That means some error message is shown to the user..
I am also not sure that NDEBUG is defined by all package maintainers. Do we even define NDEBUG in our official windows installation? I didn't do it for cppcheck premium at least.

Reply:
> I have the feeling that a `throw InternalError` would be nicer than an assert. That means some error message is shown to the user..

The user should never encounter this - that's why it is an assert. Throwing an assert might also require additional error handling as the exception might not be properly handled in some cases and that is not intended - it should still use the fallback and not error out.

The asserts are also not supposed to stay and I hope I can remove most of them after the final round of cleanups (there's at least three different PRs to apply first).

> I am also not sure that NDEBUG is defined by all package maintainers. Do we even define NDEBUG in our official windows installation? I didn't do it for cppcheck premium at least.

That would have led to previous issues with assert being hit by regular users and so I have not seen such reports.

Windows projects have this set by default so that should be the one that is correct.

Reply:
FYI there already is a ticket about `NDEBUG`: https://trac.cppcheck.net/ticket/10736.